parsec/scheduling.c

```diff
@@ -179,13 +179,14 @@ int __parsec_execute( parsec_execution_stream_t* es,
         rc = hook( es, task );
 #if defined(PARSEC_PROF_TRACE)
         task->prof_info.task_return_code = rc;
-        PARSEC_PINS(es, EXEC_END, task);
 #endif
         if( PARSEC_HOOK_RETURN_NEXT != rc ) {
             if( PARSEC_HOOK_RETURN_ASYNC != rc ) {
                 /* Let's assume everything goes just fine */
                 task->status = PARSEC_TASK_STATUS_COMPLETE;
             }
+            /* Record EXEC_END event only for incarnation that succeeds */
+            PARSEC_PINS(es, EXEC_END, task);
```

Comment:
Not sure this is what we need. As an example recursive tasks will now have an ending that is based on the moment they return, not on the moment they really complete. Thus, my first reaction was to move the `EXEC_END` event in the `if` above. But if we do this then `EXEC_END` should measure the time the runtime think a task took, so we will need to generate the same event when a task is brought back from the ASYNC mode.

Reply:
> recursive tasks will now have an ending that is based on the moment they return, not on the moment they really complete.

This is the current behavior; I thought it was intended and so maintained it. If we want to change this behavior then `EXEC_END` will need to be recorded in `__parsec_complete_execution`. It's a bit misleading to do that though, since that would mean that the "duration" of a recursive task and other asynchronous tasks includes time spent in the PaRSEC runtime or doing other work. See #396 for additional discussion on this topic.